### PR TITLE
fix scrolling when modal is open

### PR DESCRIPTION
**Pull request by FrostEvil**

### CODE REVIEW

File: src/pages/index.astro [hunk 1]
```diff
@@ -56,7 +56,7 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
   title="My portfolio"
   description="Portfolio presents my current programming skills, knowledge and stack"
 >
-  <main class="flex flex-col justify-center items-center">
+  <main class="flex flex-col justify-center items-center merro">
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:59]: `merro`?

File: src/pages/index.astro [hunk 2]
```diff
@@ -67,7 +67,7 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
         </p>
         <button class="hero--cta">Check offer</button>
       </div>
-      <div id="modal" class="modal-wrapper">
+      <div id="modal" class="modal-wrapper z-10">
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:70]: Are you sure this is the best place in the document for a modal?

File: src/scripts/modal.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
 function modal() {
   const heroContent = document.querySelector(".hero__content");
+  const projectContent = document.querySelector(".projects");
```
Comment (aleksanderwalczuk) [inline src/scripts/modal.js:3]: Not the best idea. Whenever we add a section you would add it's element to the modal script?

File: src/pages/index.astro (hunk 2 above)
Comment (aleksanderwalczuk) [reply src/pages/index.astro:70]: Beside that, I think we should have a separate component for the modal to encapsulate it's logic. Quick refactor might be good in this case

File: src/scripts/modal.js [hunk 4]
```diff
@@ -19,16 +23,20 @@ function modal() {
   }
 
   function open() {
+    modalContent.style.marginTop = `${document.documentElement.scrollTop}px`;
```
Comment (aleksanderwalczuk) [inline src/scripts/modal.js:26]: No the best solution, we can make it happen with CSS!

Comment (aleksanderwalczuk) [reply src/scripts/modal.js:26]: @FrostEvil what about this line?

File: src/scripts/modal.js (hunk 3 above)
Comment (aleksanderwalczuk) [reply src/scripts/modal.js:3]: Any ideas?

File: src/components/Modal.astro [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+<div
+  id="modal"
+  class="fixed hidden top-1/2 left-1/2 max-w-2xl w-full -translate-x-1/2 -translate-y-1/2"
+  data-show="false"
+>
+  <div
+    class="mx-8 p-6 relative flex flex-col bg-sky-200 border-sky-400 border-2 md:p-10"
+  >
+    <div class="flex justify-center">
+      <button class="p-2 absolute w-12 h-12 top-0 right-0" id="icon-close">
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:23]:
```suggestion
      <button class="btn-close p-2 absolute w-12 h-12 top-0 right-0" id="icon-close">
```

File: src/components/Modal.astro [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      </slot>
+    </div>
+    <div>
+      <slot>
+        <p class="mt-8 text-base overflow-auto max-h-[40vh] md:text-xl">
+          {description}
+        </p>
+      </slot>
+      <div class="mt-10 flex justify-between w-full">
+        <button class="btn" id="btn-close">close</button>
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:38]:
```suggestion
        <button class="btn btn-close" type="button">close</button>
```

File: src/components/Modal.astro [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        <button class="btn">confirm</button>
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:39]:
```suggestion
        <button class="btn" type="submit">confirm</button>
```

File: src/components/Modal.astro [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+---
+
+<div
+  id="modal"
+  class="fixed hidden top-1/2 left-1/2 max-w-2xl w-full -translate-x-1/2 -translate-y-1/2"
+  data-show="false"
+>
+  <div
+    class="mx-8 p-6 relative flex flex-col bg-sky-200 border-sky-400 border-2 md:p-10"
+  >
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:21]:
```suggestion
<div class="fixed inset-0 hidden" data-visible="false" id="modal">
  <div class="absolute inset-0 backdrop-blur-sm z-10"></div>
  <div class="absolute inset-0 z-20 flex justify-center items-center">
    <div class="max-w-2xl w-full">
      <div
        class="modal-container mx-8 p-6 relative flex flex-col bg-sky-200 border-sky-400 border-2 md:p-10"
      >
```

Comment (aleksanderwalczuk) [reply src/components/Modal.astro:21]: Notice the `.modal-container`, you'll need it for onOutsideClick

File: src/components/Modal.astro [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        <button class="btn" id="btn-close">close</button>
+        <button class="btn">confirm</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
+  function modal() {
+    const modalContent: HTMLElement = document.querySelector("#modal")!;
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:47]:
```suggestion
const modal = document.querySelector<HTMLElement>("#modal")!;
```

Comment (aleksanderwalczuk) [reply src/components/Modal.astro:47]: Notice how the type is being passed

Comment (aleksanderwalczuk) [reply src/components/Modal.astro:47]: Fix the `id` naming

File: src/components/Modal.astro [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    const checkOfferBtn: any = document.querySelector("#btn-offer");
+    const closeModalBtn = modalContent.querySelector("#btn-close")!;
+    const closeModalIcon = modalContent.querySelector("#icon-close")!;
+    const main = document.querySelector("#main");
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:51]: To be removed

File: src/components/Modal.astro [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    </div>
+  </div>
+</div>
+
+<script>
+  function modal() {
+    const modalContent: HTMLElement = document.querySelector("#modal")!;
+    const checkOfferBtn: any = document.querySelector("#btn-offer");
+    const closeModalBtn = modalContent.querySelector("#btn-close")!;
+    const closeModalIcon = modalContent.querySelector("#icon-close")!;
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:50]: Both are buttons and have the same purpose:
```suggestion
const closeModalBtns =
      wrapper.querySelectorAll<HTMLButtonElement>(".btn-close");
```

File: src/components/Modal.astro [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    const main = document.querySelector("#main");
+
+    function close() {
+      modalContent.dataset.show = "false";
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:54]: honestly I think `dataset.visible` would be better

File: tailwind.config.mjs [hunk 13]
```diff
@@ -1,8 +1,15 @@
...
-	plugins: [],
-}
+  content: ["./src/**/*.{astro,html,js,jsx,md,mdx,svelte,ts,tsx,vue}"],
+  theme: {
+    extend: {},
+    container: {
+      center: true,
+      padding: "1.5rem",
+      screens: {
+        lg: "1200px",
```
Comment (aleksanderwalczuk) [inline tailwind.config.mjs:10]: I'd say this is `xl`, lg should be `1024px`

File: src/components/Modal.astro [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+const {
+  modalContent: { title, description },
+} = Astro.props;
+---
+
+<div
+  id="modal"
+  class="fixed hidden top-1/2 left-1/2 max-w-2xl w-full -translate-x-1/2 -translate-y-1/2"
+  data-show="false"
+>
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:18]: All you need:
```suggestion
<div class="fixed inset-0 hidden" data-visible="false" id="modal">
```

File: src/components/Modal.astro [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+        <button class="btn">confirm</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
+  function modal() {
+    const modalContent: HTMLElement = document.querySelector("#modal")!;
+    const checkOfferBtn: any = document.querySelector("#btn-offer");
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:48]: This naming (and type :D) makes no sense. Call it a `trigger`

Comment (aleksanderwalczuk) [reply src/components/Modal.astro:48]: _When user click on a **trigger** an event is fired and a function is being called_

File: src/components/Modal.astro [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+      checkOfferBtn.disabled = true;
+      main?.classList.add("blur");
+      document.addEventListener("click", closeOutside, true);
+    }
+
+    checkOfferBtn.addEventListener("click", open);
+
+    [closeModalBtn, closeModalIcon].forEach((closeModal) =>
+      closeModal.addEventListener("click", close)
+    );
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:77]: When https://github.com/bright-bounty/astro-lp/pull/14/files#r1605886993 is applied you can just iterate on the list
```suggestion
    closeModalBtns.forEach((closeModal) =>
      closeModal.addEventListener("click", close)
    );
```

File: src/components/Modal.astro (hunk 5 above)
Comment (FrostEvil) [reply src/components/Modal.astro:23]: changed

File: src/components/Modal.astro (hunk 6 above)
Comment (FrostEvil) [reply src/components/Modal.astro:38]: changed, also removed ID

File: src/components/Modal.astro (hunk 7 above)
Comment (FrostEvil) [reply src/components/Modal.astro:39]: changed

File: src/components/Modal.astro (hunk 8 above)
Comment (FrostEvil) [reply src/components/Modal.astro:21]: changed

File: src/components/Modal.astro (hunk 9 above)
Comment (FrostEvil) [reply src/components/Modal.astro:47]: Changed

File: src/components/Modal.astro (hunk 10 above)
Comment (FrostEvil) [reply src/components/Modal.astro:51]: Removed closeModalsBtn and main

File: src/components/Modal.astro (hunk 11 above)
Comment (FrostEvil) [reply src/components/Modal.astro:50]: Changed

File: src/components/Modal.astro (hunk 12 above)
Comment (FrostEvil) [reply src/components/Modal.astro:54]: Changed

File: tailwind.config.mjs (hunk 13 above)
Comment (FrostEvil) [reply tailwind.config.mjs:10]: Changed

File: src/components/Modal.astro (hunk 14 above)
Comment (FrostEvil) [reply src/components/Modal.astro:18]: Changed

File: src/components/Modal.astro (hunk 15 above)
Comment (FrostEvil) [reply src/components/Modal.astro:48]: Changed

File: src/components/Modal.astro (hunk 16 above)
Comment (FrostEvil) [reply src/components/Modal.astro:77]: Changed

File: src/pages/index.astro [hunk 17]
```diff
@@ -140,5 +118,14 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
       </div>
     </section>
   </main>
-  <script src="../scripts/modal.js"></script>
+  <Modal modalContent={modalContent}>
```
Comment (aleksanderwalczuk) [inline src/pages/index.astro:121]: Do we actually need to pass both props and slots? I think we can the props we pass

File: src/components/Modal.astro [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+    triggerBtn.addEventListener("click", open);
+
+    closeModalBtns.forEach((closeModal) =>
+      closeModal.addEventListener("click", close)
+    );
+  }
+  document.addEventListener("DOMContentLoaded", modal);
+</script>
+
+<style>
```
Comment (aleksanderwalczuk) [inline src/components/Modal.astro:79]: We could improve this a bit
```suggestion
<style scoped>
```
https://docs.astro.build/en/guides/styling/#scoped-styles

File: src/pages/index.astro (hunk 17 above)
Comment (FrostEvil) [reply src/pages/index.astro]: pass only props

**Review by aleksanderwalczuk — APPROVED** [review]
Well done 👏

**Pull request merged**